.travis.yml

```diff
@@ -114,16 +114,15 @@ matrix:
     - ARROW_TRAVIS_OPTIONAL_INSTALL=1
     - ARROW_CPP_BUILD_TARGETS="gandiva-all"
     - ARROW_TRAVIS_USE_TOOLCHAIN=1
-    # ARROW-3979 temporarily disabled.
-    - ARROW_TRAVIS_VALGRIND=0
+    - ARROW_TRAVIS_VALGRIND=1
     - ARROW_BUILD_WARNING_LEVEL=CHECKIN
     - MATRIX_EVAL="CC=gcc-4.9 && CXX=g++-4.9"
     before_script:
     # Run if something changed in CPP or Java.
     - if [ $ARROW_CI_CPP_AFFECTED != "1" ] && [ $ARROW_CI_JAVA_AFFECTED != "1" ]; then exit; fi
     - $TRAVIS_BUILD_DIR/ci/travis_install_linux.sh
     - $TRAVIS_BUILD_DIR/ci/travis_install_clang_tools.sh
-    - $TRAVIS_BUILD_DIR/ci/travis_before_script_cpp.sh --only-library
+    - $TRAVIS_BUILD_DIR/ci/travis_before_script_cpp.sh
```

Comment:
Removing `--only-library` probably makes building slower. Is it necessary?

Reply:
This will go away with ARROW-3803

Reply:
Without this option, gandiva tests were not running as part of CI. I guess in the travis gandiva script, gandiva tests option is enabled only if "--only-library" was not enabled. But, if the 3803 takes care of it, I am fine removing it.